server/metrics.go

```diff
@@ -148,10 +155,61 @@ func TimedAndCounted(handler http.Handler, fullPath string, method string, p pro
 	}
 }
 
-// PrometheusTimedAndCounted wraps a http.Handler with via prometheus.InstrumentHandler
+// PrometheusTimedAndCounted wraps a http.Handler with via promhttp.InstrumentHandlerCounter and promhttp.InstrumentHandlerDuration
 func PrometheusTimedAndCounted(handler http.Handler, name string) *Timer {
 	return &Timer{
-		isProm:  true,
-		handler: prometheus.InstrumentHandler(name, handler),
+		isProm: true,
+		handler: promhttp.InstrumentHandlerCounter(prometheusCounter(name),
```

Comment:
This is a terrible idea. It'll register a new *metric* for every endpoint instead of using labels and one metric called `http_requests_total` with the label `handler` set to the respective endpoint.

On top of that it fails for any endpoint with a `.` in the URL (like e.g. `/broken.json`) as the dot is an invalid character in the metric name.